### PR TITLE
Adjust the location of the signer snippet

### LINKED ISSUE #3867: Provide means of adjusting the location of the signer snippet
Currently fixed at bottom right. Would be nice to at least have the option of moving to top-right.

### PR DESCRIPTION
**Pull request by ngotchac**
Closes #3867 

Add a small button on the signer tab to drag it. The signer opens accordingly to the position.

output

### CODE REVIEW

**Comment by jacogr** [comment]
Keep the related https://github.com/ethcore/parity/issues/3941 in mind.

Would prefer on-ice until the above is in since this approach causes some issues with a larger & needed expansion. 

Think the approach is over-the-top atm (although nice), the dapp being to specify position in the manifest (top-left, top-right, bottom-left, bottom-right) would have been sufficient and consistent. The reality is that for one specific dapp it would need to be dragged and then back again in others. (As to not overlap with the UI - so users would need to repeat each time they swap between affected and non-affected apps)

**Comment by gavofyork** [comment]
agree with jaco here; that said, it might be nice to explore how this functionality could work with the dapp-specified position.

**Comment by ngotchac** [comment]
We could also save the location per dapp in local storage

**Comment by gavofyork** [comment]
yeah; can we alter this to allow the dapp to specify a default position?

**Comment by ngotchac** [comment]
@gavofyork Ok so now the position of the Parity Bar is saved in local storage per dapp ; and each dapp can have a default position within `top-right`, `top-left`, `bottom-right` and `bottom-left`.

**Comment by gavofyork** [comment]
conflicts..

File: js/src/views/Dapps/dappsStore.js [hunk 1]
```diff
@@ -44,13 +45,27 @@ export default class DappsStore {
...
     this.loadExternalOverlay();
     this.subscribeToChanges();
   }
 
+  static get (api) {
+    if (!instance) {
+      instance = new DappsStore(api);
+    }
+
+    if (api && !instance._api) {
```
Comment (jacogr) [inline js/src/views/Dapps/dappsStore.js]: Not sure why we are doing this - when we create it we should properly pass in the api. (There is one in the app, no reason why it cannot be passed in)

**Comment by jacogr** [comment]
I'm honestly still not even close to on-board with the manua moving. 

It really is going to create issues for the next phase where the expanded ParityBar moves to a larger Portal (either opened from top/bottom depending on where base is located.) It does work with the current small expanded view, but cannot see how to do things properly when the bar is moved to the middle, unless it is just ignored.

Lots of code to move stuff around that adds maintenance complexity as well and I don't see how it really adds value to the end-user apart from "cool to have".

On the plus side, do think the per-dapp specification as it is no implemented works well.

**Comment by ngotchac** [comment]
Well it doesn't really get it the middle : the expanded signer can be in four position (computed automatically based on the small parity bar) : NE / NW / SE / SW
The Parity Bar (not expanded) is always whether on top or on bottom of the screen. The user can choose where to put it on the horizontal axis.

Regarding migrating to Portal, I don't think that's really an issue. The part of the code that will get modified is the expanded view, which hasn't been touched.

As everything is already written, I don't think it would add any more work to port the Signer to the Portal view (compared with porting to the Portal view from current master)

**Comment by ngotchac** [comment]
As spoken with @jacogr , the Parity Bar now sticks to only one of the four corners of the screen.

**Pull request merged**